test/integration/scheduler/util.go

```diff
@@ -322,6 +322,39 @@ func waitForNodeLabels(cs clientset.Interface, nodeName string, labels map[strin
 	return wait.Poll(time.Millisecond*100, wait.ForeverTestTimeout, nodeHasLabels(cs, nodeName, labels))
 }
 
+// nodeHasImages returns a function that checks if a node has all the given images.
+func nodeHasImages(cs clientset.Interface, nodeName string, images []imageutils.ImageConfig) wait.ConditionFunc {
+	return func() (bool, error) {
+		node, err := cs.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		if err != nil {
+			// This could be a connection error so we want to retry.
+			return false, nil
+		}
+
+		nodeImages := make(map[string]struct{})
```

Comment:
There's a `"k8s.io/apimachinery/pkg/util/sets"` if you want to have a cleaner code. Not mandatory anyway.

Reply:
addressed